ui/src/app/settings/components/certs-list/certs-list.tsx

```diff
@@ -0,0 +1,253 @@
+                        </DataLoader>
+                    </div>
+                </div>
+                <SlidingPanel isShown={this.showAddTLSCertificate} onClose={() => this.showAddTLSCertificate = false} header={(
+                    <div>
+                        <button className='argo-button argo-button--base' onClick={() => this.formApiTLS.submitForm(null)}>
+                            Create
+                        </button>
+                        <button onClick={() => this.showAddTLSCertificate = false}
+                                className='argo-button argo-button--base-o'>
+                            Cancel
+                        </button>
+                    </div>
+                )}>
+                    <h4>Create TLS repository certificate</h4>
+                    <Form onSubmit={(params) => this.addTLSCertificate(params as NewTLSCertParams)}
+                          getApi={(api) => this.formApiTLS = api}
+                          preSubmit={(params: NewTLSCertParams) => ({
+                              servername: params.servername,
+                              certdata: btoa(params.certdata),
+                          })}
+                          validateError={(params: NewTLSCertParams) => ({
+                              servername: !params.servername && 'Repository server name is required',
+                              certdata: !params.certdata && 'Certificate data is required',
+                          })}>
+                        {(formApiTLS) => (
+                            <form onSubmit={formApiTLS.submitForm} role='form' className='certs-list width-control' encType='multipart/form-data'>
+                                <div className='argo-form-row'>
+                                    <FormField formApi={formApiTLS} label='Repository server name' field='servername' component={Text}/>
+                                </div>
+                                <div className='argo-form-row'>
+                                    <FormField formApi={formApiTLS} label='TLS certificate (PEM format)' field='certdata' component={TextArea}/>
+                                </div>
+                           </form>
+                        )}
+                    </Form>
+                </SlidingPanel>
+                <SlidingPanel isShown={this.showAddSSHKnownHosts} onClose={() => this.showAddSSHKnownHosts = false} header={(
+                    <div>
+                        <button className='argo-button argo-button--base' onClick={() => this.formApiSSH.submitForm(null)}>
+                            Create
+                        </button>
+                        <button onClick={() => this.showAddSSHKnownHosts = false}
+                                className='argo-button argo-button--base-o'>
+                            Cancel
+                        </button>
+                    </div>
+                )}>
+                    <h4>Create SSH known host entries</h4>
```

Comment:
Minor issue: known hosts form input is not erased after clicking 'Create'/'Cancel' button . Simplest way to fix it is to don't render form when panel is closed:

```
{this.showAddSSHKnownHosts && <React.Fragment>
    <h4>Create SSH known host entries</h4>
    ....
<React.Fragment>}
```

Same with TLS cert form

Reply:
Didn't get your proposed solution to work, so I added a `private clearForms()` method which gets called on visibility change and just sets data to empty strings for all fields.